### PR TITLE
Flowc single element struct

### PR DESCRIPTION
**Pull request by dmitry-vlasov**

### CODE REVIEW

File: tools/flowc/flowc_usage.flow [hunk 1]
```diff
@@ -199,6 +199,9 @@ printUsage(config) {
 	prih(help,"      inline-max-nesting=<int>  The boundary for nesting level of primitive calls in functions to be inlined");
 	prih(help,"      do-const-structs=1        Inline variables of limited nesting, which refer to constant structs");
 	prih(help,"      remove-dup-strings=1      Global optimization: remove duplicate string literals with variables, holding common strings");
+	prih(help,"      remove-single-structs=1   Remove structs with only one immutable argument, which are not contained in any union");
```
Comment (alstrup) [inline tools/flowc/flowc_usage.flow:202]: It is not removing, but unfolding or inlining, right?

File: tools/flowc/manipulation/transform_programs.flow [hunk 2]
```diff
@@ -60,6 +60,9 @@ export {
 
 	// Remove sequences with exactly one element
 	fiRemoveSigletonSeqences(p: FiProgram) -> FiProgram;
+
+	// Remove single element structs, which are not in unions
+	fiRemoveSigleElementStructs(p: FiProgram) -> FiProgram;
```
Comment (alstrup) [inline tools/flowc/manipulation/transform_programs.flow:65]: Remove is not the right term. Misspelling single.

File: tools/flowc/manipulation/transform_programs.flow [hunk 3]
```diff
@@ -1018,3 +1021,209 @@ fiRemoveSigletonSeqences(p: FiProgram) -> FiProgram{
...
+		fold(
+			filtermap(strSplit(to_remove_param, ","), \name -> if (name == "") None() else Some(name)),
+			makeTree(),
+			\acc, name -> switch (lookupTree(p.names.structs, name)) {
+				Some(s): if (is_single_field(s)) setTree(acc, name, s) else acc;
+				None(): acc;
+			}
+		);
+	}
+	if (isConfigParameterTrue(p.config.config, "dump-single-element-structs")) {
```
Comment (alstrup) [inline tools/flowc/manipulation/transform_programs.flow:1067]: The dump switch is not documented in help, but useful

File: tools/flowc/flowc_usage.flow (hunk 1 above)
Comment (dmitry-vlasov) [reply tools/flowc/flowc_usage.flow:202]: Currently it's completely removing from program.

File: tools/flowc/manipulation/transform_programs.flow (hunk 2 above)
Comment (dmitry-vlasov) [reply tools/flowc/manipulation/transform_programs.flow]: Fixed.

**Pull request merged**